### PR TITLE
Introduced a default section for deprecated modules

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Some modules need to be updated because they are registered into old tabs that doesnt exists anymore. For now, they will be put in a new section called "DEFAULT". Note this section _will be removed_ in 1.7.1 |
| Type? | new feature |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | yes |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-711 |
| How to test? | You can install Express Mailing module and see what happens :) |

### CODE REVIEW

**Comment by Shudrum** [comment]
If I can:

The last commit should be named `TE: selenium tests are now executed in prod environment`

**Comment by mickaelandrieu** [comment]
Yeah, make sense :) Thank you for review @Shudrum

File: classes/Tab.php [hunk 1]
```diff
@@ -290,6 +290,21 @@ public static function getIdFromClassName($class_name)
     }
 
     /**
+     * @deprecated Deprecated since 1.7.0, to be removed in 1.7.1. Upgrade module to 1.7.
+     */
+    private static function getClassName($class_name)
+    {
+        $legacyClassNames = array('AdminTools', 'AdminPriceRule', 'AdminAdmin', 'AdminParentStats', 'AdminMarketing', 'AdminCarrierWizard', 'AdminQuickAccesses', 'AdminTabs' );
```
Comment (Shudrum) [inline classes/Tab.php]: Please change to:

```php
$legacyClassNames = array(
    'AdminTools',
    [...]
    'AdminTabs',
);
```

File: classes/Tab.php [hunk 2]
```diff
@@ -290,6 +290,21 @@ public static function getIdFromClassName($class_name)
...
+
+        if(in_array($class_name, $legacyClassNames)) {
+            @trigger_error($class_name. ' is a deprecated tab since version 1.7.0 and "Default" will be removed in 1.7.1.. Upgrade module using the docs: http://build.prestashop.com/news/how-we-reorganized-main-menu-prestashop-1.7/.', E_USER_DEPRECATED);
```
Comment (Shudrum) [inline classes/Tab.php]: 👍  I will add the waiting to wording tag.

File: install-dev/data/xml/tab.xml [hunk 3]
```diff
@@ -410,6 +410,9 @@
                 <tab id="Multistore" id_parent="Advanced_Parameters" active="0" hide_host_mode="0" icon="">
                     <class_name>AdminShopGroup</class_name>
                 </tab>
-
+      <!--DEFAULT-->
+      <tab id="DEFAULT" id_parent="" active="1" hide_host_mode="0" icon="">
+        <class_name>DEFAULT</class_name>
```
Comment (Shudrum) [inline install-dev/data/xml/tab.xml]: Yep, it need some wordings AND localizations.

**Comment by Shudrum** [comment]
Please add the class and title on all languages tabs to this pull request.

**Comment by mickaelandrieu** [comment]
@Shudrum done.

**Pull request merged**